DeviceIOContoller/CPP/README.md

```diff
@@ -0,0 +1,282 @@
+</tr>
+<tr>
+    <td>"D:P(A;;GA;;;SY)"</td>
+    <td>Denies access to all, except kernel mode code, and usermode clients running under the SYSTEM account.</td>
+    <td>SDDL_DEVOBJ_SYS_ALL</td>
+</tr>
+<tr>
+    <td>"D:P(A;;GA;;;SY)(A;;GRGWGX;;;BA)(A;;GR;;;WD)"</td>
+    <td>Grants full access to kernel mode clients and the SYSTEM account, grants read, write, execute to built-in Administrators, and grants read-only access to everyone else.</td>
+    <td>SDDL_DEVOBJ_SYS_ALL_ADM_RWX_WORLD_R</td>
+</tr>
+</tbody>
+</table>
+
+A number of common security descriptor strings are defined in wdmsec.h.
+
+To make a device accessible to UWP applications, you need to grant access to the All Application Packages security principle, which is represented by the SID string "AC".  The following SDDL will grant GENERIC_ALL permissions to SYSTEM, built-in administrators, authenticated users, and appcontainer applications:
+
+    "D:P(A;;GA;;;SY)(A;;GA;;;BA)(A;;GA;;;AU)(A;;GA;;;AC)"
+
+If you control the code of the driver you wish to access, you can programatically assign an SDDL string to a device object using the [IoCreateDeviceObjectSecure()](https://msdn.microsoft.com/en-us/library/windows/hardware/ff548407(v=vs.85).aspx) routine in WDM drivers, or the [WdfDeviceInitAssignSDDLString()](https://msdn.microsoft.com/en-us/library/windows/hardware/ff546035(v=vs.85).aspx) routine in WDF drivers.
+
+If you do not control the driver code for your device, you can override the SDDL for any device using the registry. If a registry value named "Security" is present in the hardware registry key for a device, the system will use the specified security descriptor instead of the one supplied by the driver. We will override the security descriptor for the COM port with one that grants access to all appcontainer applications.
+
+First, we need to convert the SDDL string to its binary representation. We can do this with [PowerShell](http://blogs.technet.com/b/heyscriptingguy/archive/2011/08/09/use-powershell-to-convert-sddl-to-binary-format.aspx). Open a PowerShell prompt and run the following commands:
+
+```csharp
+    $a = ([wmiclass]"Win32_SecurityDescriptorHelper").SDDLToBinarySD("D:P(A;;GA;;;SY)(A;;GA;;;BA)(A;;GA;;;AU)(A;;GA;;;AC)").BinarySD | % { '{0:X}' -f $_ }
+    $a -join ','
+```
+
+The output will be a long string of comma-separated numbers that we will paste into a registry file. Create a file named deviceac.reg with the following contents:
+
+    Windows Registry Editor Version 5.00
+
+    [HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Enum\ACPI\PNP0501\1]
+    "Security"=hex:1,0,4,90,0,0,0,0,0,0,0,0,0,0,0,0,14,0,0,0,2,0,60,0,4,0,0,0,0,0,14,0,0,0,0,10,1,1,0,0,0,0,0,5,12,0,0,0,0,0,18,0,0,0,0,10,1,2,0,0,0,0,0,5,20,0,0,0,20,2,0,0,0,0,14,0,0,0,0,10,1,1,0,0,0,0,0,5,B,0,0,0,0,0,18,0,0,0,0,10,1,2,0,0,0,0,0,F,2,0,0,0,1,0,0,0
+
+Copy this file to c:\data on your MBM and run:

```

Comment:
This is referencing Minnow Board Max (MBM).  It needs to be updated for UP Board.

Reply:
Content for MBM is removed